File: python/queue.py
class Queue:

  def __init__(self):
    self.items = {};
    self.size = 0;


  def enqueue(self, item):
    if self.size == 0:
      self.items[0] = item;
      self.size = self.size + 1;
    else:
      all_items = self.items.values();
      all_items = [item] + all_items;
      self.size = self.size + 1;

      for i in range(len(all_items)):
        self.items[i] = all_items[i];


  def dequeue(self):
    first_in_queue = '';
    all_items = self.items.values();
    first_in_queue = all_items[len(all_items) - 1];
    upto = len(all_items) - 1;
    ze_items = all_items[0:upto];

    for i in range(len(ze_items)):
      self.items[i] = ze_items[i];

    del self.items[(self.size - 1)];
    self.size = self.size - 1;

    if type(first_in_queue) == int:
      print("FIRST IN QUEUE IS {:d}".format(first_in_queue));
    else:
      print("FIRST IN QUEUE IS {:s}".format(first_in_queue));
    return first_in_queue;


  def contains(self, item):
    print(len(list(filter(lambda x: x == item, self.items.values()))) > 0)
    return len(list(filter(lambda x: x == item, self.items.values()))) > 0;



  def clear(self):
    self.items = {};
    self.size = 0;


  def peek(self):
    print(self.items[self.size - 1]);
    return self.items[self.size - 1];


  def to_array(self):
    print(self.items.values());
    return self.items.values();




abc = Queue();
print(abc);
abc.enqueue(1);
abc.enqueue(2);
abc.enqueue(3);
print("\n\n\n\n", abc)
abc.dequeue();
print(abc.items)
abc.contains(2);
abc.peek();
abc.to_array();

File: go/queue.go
package main

import (
	"fmt"
	"strconv"
)

type Queue struct {
	ItemsInfo  string
	Items      [][]string
	ItemsComma string
	SizeInfo   string
	Size       int
}

func InitQueue() *Queue {
	var items = make([][]string, 1)
	return &Queue{
		ItemsInfo:  "items:",
		Items:      items,
		ItemsComma: ",",
		SizeInfo:   "size:",
		Size:       0,
	}
}

func (q *Queue) Enqueue(item string) {
	if q.Size == 0 {
		key := strconv.Itoa(0)
		q.Items[0] = []string{key, item}
		q.Size += 1
	} else {
		var allItems [][]string
		var newAllItems [][]string
		allItems = make([][]string, 0)

		for i := 0; i < len(q.Items); i++ {
			if len(q.Items[i]) == 2 {
				allItems = append(allItems, q.Items[i])
			}
		}

		newItemIndex := strconv.Itoa(q.Size)
		newAllItems = make([][]string, 0)
		newAllItems = append(newAllItems, []string{newItemIndex, item})

		allItems = append(newAllItems, allItems...)

		q.Size += 1
		q.Items = allItems
	}
}

func first(args ...interface{}) interface{} {
	return args[0]
}

func (q *Queue) Dequeue() []string {
	var poppedItem []string
	var newItems [][]string
	poppedItem = q.Items[len(q.Items)-1]
	q.Items = q.Items[0 : len(q.Items)-1]
	newItems = make([][]string, 0)

	for i := 0; i < len(q.Items); i++ {
		newItems = append(newItems, []string{strconv.Itoa(len(q.Items) - (i + 1)), q.Items[i][1]})
	}

	q.Size -= 1
	q.Items = newItems

	fmt.Printf("%s\n", poppedItem)
	return poppedItem
}

func (q *Queue) Clear() {
	q.Size = 0
	q.Items = make([][]string, 1)
}

func (q *Queue) IsEmpty() bool {
	fmt.Println(q.Size == 0)
	return q.Size == 0
}

func (q *Queue) QueueSize() int {
	fmt.Println(q.Size)
	return q.Size
}

func main() {
	oo := InitQueue()
	oo.Enqueue("hello")
	oo.Enqueue("1234")
	oo.Enqueue("3344")
	oo.Enqueue("52345")
	oo.Dequeue()
	fmt.Println(oo.Items)
	oo.Enqueue("Alex")
	oo.Enqueue("John")
	oo.Enqueue("Computer Science")
	oo.Enqueue("RANDOM INPUT")
	fmt.Println(oo.Items)
}
